### PR TITLE
Feature move sessions 2 sessionmaker

### PR DESCRIPTION
**Pull request by il-ia-ni**
Hi Phillip :) I was finally able to open my first PR to this repo!

In this branch I have implemented the central engine and session instances in the Main-method using the db_connection Script provided by you.

I have also introduced 3 ORM-Classes for test tables of my local DB as well as a basic loguru-formatter that saves logs in files locally on a disk. 

Finally, I have adjusted my first sql-requests with SQL-Alchemy to work with the central connection instances.

### CODE REVIEW

**Review by Phillip-Oliver — COMMENTED** [review]
Kannst du hieraus mal einen separate Methode machen ?
grafik
?
Und dann vllt: 
-url_SQLSERVERTestDBMS
-echo
-echo_pool=True
-future=True

Als Parameter der Funktion festlegen

File: ddl_scripts/adding_data.py [hunk 1]
```diff
@@ -1,25 +1,46 @@
...
+    id="test_obj_1",
+    create_date=datetime.utcnow(),
+    update_date=datetime.utcnow(),
+    name="Test signal meta",
+    caster_id="testCaster",
+    category="test_category",
+    external_id="TEST.testCaster",
+    units="TestUnit",
+    description="This is a test object",
+)
```
Comment (Phillip-Oliver) [inline ddl_scripts/adding_data.py:32]: Gefällt mir gut ! Würde aber versuchen, daraus eine Methode/Funktion zu machen :)

File: dql_scripts/select_joins.py [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
+session1 = Session(engine_SQLServerTest_MainDB)
 
 select_join_orm_stmt1 = (
     # JOIN with left and right explicit sides + automatic ON-clause
     # See https://docs.sqlalchemy.org/en/14/tutorial/data_select.html#explicit-from-clauses-and-joins
     select(DefectRootCause)
     .join_from(DefectRootCause, DefectEvent)
     .join_from(DefectRootCause, SignalMeta)
-    .where(DefectRootCause.event_id == 2407113)
+    .where(DefectRootCause.event_id == 2407113)  # TODO: Upd join stmts to have no where-clauses for pandas filtering
```
Comment (Phillip-Oliver) [inline dql_scripts/select_joins.py:20]: Hier könntest du mal schauen, wie nach mehrer event_ids gleichzeitig Filtern könntest ;)

File: dql_scripts/select_joins.py [hunk 3]
```diff
@@ -32,12 +36,15 @@
...
 
-def get_select_join_orm_result(select_stmt):
-    result = session.scalars(select_stmt).all()
-    for ormObj in result:
-        print(ormObj)
+def get_select_join_orm_result(session: sqlalchemy.orm.session, select_stmt):
+    with SessionTestSQLServer.begin() as session3:
+        logger.info(f"Choosing joined data with following select-statement: {select_stmt}")
+
+        result = session3.scalars(select_stmt).all()  # TODO: Also make with .execute for using with pandas
```
Comment (Phillip-Oliver) [inline dql_scripts/select_joins.py:47]: Was wäre hier der Unterscheid zwischen .all() und .fetchall() ?

**Review by Phillip-Oliver — COMMENTED** [review]
Finde es gut, dass du hier diese statements als text verwendet hast !
grafik

Kannst du mal probieren, dies hier in einen text zu bekommen:
`select count(case when signal_id in ('cast_speed_1', 'mold_level_1', 'stopper_rod_1') then B.event_id end) Grp_1,
count(case when signal_id in ('cast_speed_1', 'stopper_rod_1', 'tundish_weight_1') then B.event_id end) Grp_3
from dbo.defect_event A inner join dbo.defect_root_cause B on A.event_id=B.event_id
where A.event_id in
(
select dbo.defect_event.event_id
from dbo.defect_event
inner join dbo.defect_root_cause on dbo.defect_event.event_id = dbo.defect_root_cause.event_id where strand_id = '1_1'
group by dbo.defect_event.event_id
having count(signal_id) > 2 and count(signal_id) < 4)
and behaviour_pattern_id = 'clogging'
and strand_id = '1_1'`

File: dql_scripts/select_joins.py (hunk 2 above)
Comment (il-ia-ni) [reply dql_scripts/select_joins.py:20]: `.where(
        or_(
            DefectRootCause.event_id == 2407113, DefectRootCause.event_id == 2407026, DefectRootCause.event_id == 2407022
        )
    )`

Found here: [https://docs.sqlalchemy.org/en/14/tutorial/data_select.html#the-where-clause](https://docs.sqlalchemy.org/en/14/tutorial/data_select.html#the-where-clause)

File: dql_scripts/select_joins.py (hunk 3 above)
Comment (il-ia-ni) [reply dql_scripts/select_joins.py:47]: Laut den DocStrings ist .fetchall() bloß ein Synonym zum .all()?

File: dql_scripts/select_joins.py (hunk 2 above)
Comment (il-ia-ni) [reply dql_scripts/select_joins.py:20]: Option 2:
`join_query = select(DefectEvent.create_date,                    DefectEvent.event_type,                    DefectEvent.event_id,                    DefectEvent.caster_id,                    DefectEvent.strand_id,                    DefectEvent.grade_id,                    DefectEvent.behaviour_pattern_id,                    DefectEvent.detection_probability,                    DefectRootCause.signal_id,                    DefectRootCause.importance,                    DefectRootCause.signal_data                    ).join(DefectEvent,                           DefectRootCause.event_id == DefectEvent.event_id)\    .filter(DefectEvent.behaviour_pattern_id.in_(defects))\    .filter(DefectEvent.strand_id.in_(strand_ids))\    .filter(DefectEvent.create_date.between(start, end))`

**Review by Phillip-Oliver — APPROVED** [review]
Nice work ! 🚀

**Pull request merged**